adr/0002-quarkus-rest.adoc

```diff
@@ -0,0 +1,37 @@
+= Rename RESTEasy Reactive extensions to Quarkus REST
+
+* Status: proposed
+* Date: 2023-10-17
+
+== Context and Problem Statement
+
+RESTEasy Reactive is the preferred REST stack in Quarkus, however in the almost 3 years since its release, we have seen
+folks of all experience misunderstand its capabilities and being led to believe that it forces reactive programming on them.
+
+We have tried to clarify the situation with blog posts, talks and on the various communication channels, but there seems to be
+no escaping the fact that "reactive" in the name primes people in a non-desirable way.
+
+The proposal is to rename the Quarkus extensions only (the independent project itself would continue to be called RESTEasy Reactive)
+to `Quarkus REST`.
+
+== Considered options
+
+We can continue with the current situation, but that feels less than ideal given the recurrence of the problem.
+
+== Decision
+
+* Rename all `quarkus-resteasy-reactive*` maven modules to `quarkus-rest*`
```

Comment:
What about the reactive-rest-client? I do not believe it has quarkus-resteasy-reactive in the name.

Reply:
Indeed, but what would we call it? `quarkus-rest-client`?

Reply:
The problem is that that name is already taken by the old client that is based on RESTEasy Classic.

Reply:
That's exactly my problem. We need to find a proper new name for this one.

Reply:
It might be controversial but if we need to pave way for the future, I would rather break things and make it clear people need to point to a new artifact when using RESTEasy Classic rather than trying to find a new name that will probably be less appealing.

Reply:
I also have no good alternatives to offer, so I also (reluctantly) agree :)

Reply:
If we are afraid to break applications that are using `quarkus-rest-client` and that the code wouldn't work using the reactive client, then I think we might want to consider having a rewrite rule so that when users run `quarkus update` we will automatically rewrite the dependency to rename the `quarkus-rest-client` to whatever name we agree to use for the non-reactive client. 

We obviously want to promote the new `quarkus-rest-client`, but it might be better to have that as an active decision rather than breaking existing applications.

Reply:
Note that we are pushing more and more for people to use `quarkus update` to update to the next version. This can be easily handled by `quarkus update` if I'm not mistaken as we can change the dependencies when we update to version x.y and then you're good to go.

Fixing a bad naming issue by introducing a not ideal name is going to bite us in the future I think.

I agree not everyone will use `quarkus update` but still that's a good way to make the final push for it.